pybamm/expression_tree/binary_operators.py

```diff
@@ -115,12 +115,21 @@ def __str__(self):
             right_str = f"{self.right!s}"
         return f"{left_str} {self.name} {right_str}"
 
-    def create_copy(self):
+    def create_copy(
+        self, new_children: list[pybamm.Symbol, pybamm.Symbol] | None = None
+    ):
         """See :meth:`pybamm.Symbol.new_copy()`."""
 
         # process children
-        new_left = self.left.new_copy()
-        new_right = self.right.new_copy()
+        if new_children is None:
```

Comment:
I feel like all this logic could go in the `pybamm.Symbol.create_copy`, and this function should just check the number of children then call `pybamm.Symbol.create_copy`